coap/api/transport.go

```diff
@@ -66,75 +67,84 @@ func MakeCoAPHandler(svc coap.Service, l *slog.Logger) mux.HandlerFunc {
 	return handler
 }
 
-func sendResp(w mux.ResponseWriter, resp *message.Message) {
-	if err := w.Client().WriteMessage(resp); err != nil {
+func sendResp(w mux.ResponseWriter, resp *pool.Message) {
+	if resp.Code() == codes.Content {
+		obs, err := resp.Options().Observe()
+		if err != nil {
+			logger.Warn(fmt.Sprintf("Error reading observe option: %s", err))
+		}
+		if obs == startObserve {
+			return
+		}
+	}
+	if err := w.Conn().WriteMessage(resp); err != nil {
 		logger.Warn(fmt.Sprintf("Can't set response: %s", err))
 	}
 }
 
 func handler(w mux.ResponseWriter, m *mux.Message) {
-	resp := message.Message{
-		Code:    codes.Content,
-		Token:   m.Token,
-		Context: m.Context,
-		Options: make(message.Options, 0, 16),
-	}
-	defer sendResp(w, &resp)
+	resp := pool.NewMessage(w.Conn().Context())
+	resp.SetToken(m.Token())
+	defer sendResp(w, resp)
+
 	msg, err := decodeMessage(m)
 	if err != nil {
 		logger.Warn(fmt.Sprintf("Error decoding message: %s", err))
-		resp.Code = codes.BadRequest
+		resp.SetCode(codes.BadRequest)
 		return
 	}
 	key, err := parseKey(m)
 	if err != nil {
 		logger.Warn(fmt.Sprintf("Error parsing auth: %s", err))
-		resp.Code = codes.Unauthorized
+		resp.SetCode(codes.Unauthorized)
 		return
 	}
-	switch m.Code {
+
+	switch m.Code() {
 	case codes.GET:
-		err = handleGet(m.Context, m, w.Client(), msg, key)
+		err = handleGet(context.Background(), m, w, msg, key)
+		resp.SetCode(codes.Content)
```

Comment:
@1998-felix In my suggestion I might wronly used context.Background(), Please switch bask to m.Context()